src/representative/rep_periods.jl

```diff
@@ -0,0 +1,200 @@
+[`SingleReprPeriodWrapper`](@ref). This coresponds to the default behavior.
+"""
+repr_periods(ts::TimeStructure) = SingleReprPeriodWrapper(ts)
+
+Base.length(rpers::SingleReprPeriodWrapper) = 1
+function Base.eltype(::Type{SingleReprPeriodWrapper{T,OP}}) where {T,OP}
+    return SingleReprPeriod{T,OP}
+end
+
+function Base.iterate(rpers::SingleReprPeriodWrapper, state = nothing)
+    !isnothing(state) && return nothing
+    return SingleReprPeriod(rpers.ts), 1
+end
+Base.last(rpers::SingleReprPeriodWrapper) = SingleReprPeriod(rpers.ts)
+
+"""
+When the `TimeStructure` is a [`TwoLevel`](@ref), `repr_periods` returns an `Array` of
+all [`StratReprPeriod`](@ref)s.
+"""
+function repr_periods(ts::TwoLevel)
+    return collect(
+        Iterators.flatten(repr_periods(sp) for sp in strategic_periods(ts)),
+    )
+end
+
+"""
+    RepresentativePeriod{T,OP<:TimeStructure{T}} <: AbstractRepresentativePeriod{T}
+
+A type representing a single representative period supporting iteration over its
+time periods. It is created through iterating through [`ReprPeriods`](@ref).
+"""
+struct RepresentativePeriod{T,OP<:TimeStructure{T}} <:
+       AbstractRepresentativePeriod{T}
+    rp::Int
+    mult_rp::Float64
+    operational::OP
+end
+
+_rper(rp::RepresentativePeriod) = rp.rp
+
+mult_repr(rp::RepresentativePeriod) = rp.mult_rp
+
+Base.show(io::IO, rp::RepresentativePeriod) = print(io, "rp-$(_rper(rp))")
+function Base.eltype(_::Type{RepresentativePeriod{T,OP}}) where {T,OP}
+    return ReprPeriod{eltype(OP)}
+end
+Base.length(rp::RepresentativePeriod) = length(rp.operational)
+
+# Provide a constructor to simplify the design
+function ReprPeriod(
+    rp::RepresentativePeriod,
+    per::P,
+) where {P<:Union{TimePeriod,TimeStructure}}
+    mult = mult_repr(rp) * multiple(per)
+    return ReprPeriod(_rper(rp), per, mult)
+end
+
+# Iterate the time periods of a RepresentativePeriod
+function Base.iterate(rp::RepresentativePeriod, state = nothing)
+    next =
+        isnothing(state) ? iterate(rp.operational) :
+        iterate(rp.operational, state)
+    isnothing(next) && return nothing
+
+    return ReprPeriod(rp, next[1]), next[2]
+end
+function Base.getindex(rp::RepresentativePeriod, index::Int)
+    per = rp.operational[index]
+    return ReprPeriod(rp, per)
+end
+function Base.eachindex(rp::RepresentativePeriod)
+    return eachindex(rp.operational)
+end
+function Base.last(rp::RepresentativePeriod)
+    per = last(rp.operational)
+    return ReprPeriod(rp, per)
+end
+
+"""
+    ReprPeriods{T,OP}
+
+Iterator for iterating through the individual presentative periods of a
+[`RepresentativePeriods`](@ref) time structure. It is automatically created through the
+function [`repr_periods`](@ref).
+"""
+struct ReprPeriods{S,T,OP}
+    ts::RepresentativePeriods{S,T,OP}
+end
+
+"""
+When the `TimeStructure` is a [`RepresentativePeriods`](@ref), `repr_periods` returns the
+iterator [`ReprPeriods`](@ref).
+"""
```

Comment:
As above, I am bit skeptical about referring to `ReprPeriods` which is an internal structure. Do we need a separate docstring for this version of `repr_periods` or should it be covered by one general docstring?

Reply:
We could include it for the general. The advantage of individual is that it is easy to extend the docstring. Regarding the other points, see above.